packages/cli/src/commands/database/alteration/index.test.ts

```diff
@@ -1,42 +1,45 @@
+import { mockEsmWithActual } from '@logto/shared/esm';
+import Sinon from 'sinon';
 import { createMockPool } from 'slonik';
 
-import * as queries from '../../../queries/logto-config.js';
-import type { QueryType } from '../../../test-utilities.js';
-import * as functions from './index.js';
 import { chooseAlterationsByVersion } from './version.js';
 
-const mockQuery: jest.MockedFunction<QueryType> = jest.fn();
+const { jest } = import.meta;
 
 const pool = createMockPool({
-  query: async (sql, values) => {
-    return mockQuery(sql, values);
-  },
+  query: jest.fn(),
 });
 
-describe('getUndeployedAlterations()', () => {
-  const files = Object.freeze([
-    { filename: '1.0.0-1663923770-a.js', path: '/alterations-js/1.0.0-1663923770-a.js' },
-    { filename: '1.0.0-1663923771-b.js', path: '/alterations-js/1.0.0-1663923771-b.js' },
-    { filename: '1.0.0-1663923772-c.js', path: '/alterations-js/1.0.0-1663923772-c.js' },
-  ]);
+const files = Object.freeze([
+  { filename: '1.0.0-1663923770-a.js', path: '/alterations-js/1.0.0-1663923770-a.js' },
+  { filename: '1.0.0-1663923771-b.js', path: '/alterations-js/1.0.0-1663923771-b.js' },
+  { filename: '1.0.0-1663923772-c.js', path: '/alterations-js/1.0.0-1663923772-c.js' },
+]);
 
-  beforeEach(() => {
-    // `getAlterationFiles()` will ensure the order
-    jest.spyOn(functions, 'getAlterationFiles').mockResolvedValueOnce([...files]);
-  });
+await mockEsmWithActual('./utils.js', () => ({
+  getAlterationFiles: async () => files,
+}));
+
+const { getCurrentDatabaseAlterationTimestamp } = await mockEsmWithActual(
+  '../../../queries/logto-config.js',
+  () => ({
+    getCurrentDatabaseAlterationTimestamp: jest.fn(),
+  })
+);
+
+const { getUndeployedAlterations } = await import('./index.js');
 
+describe('getUndeployedAlterations()', () => {
   it('returns all files if database timestamp is 0', async () => {
-    jest.spyOn(queries, 'getCurrentDatabaseAlterationTimestamp').mockResolvedValueOnce(0);
+    getCurrentDatabaseAlterationTimestamp.mockResolvedValue(0);
```

Comment:
We need to clear this mock after each test is done, for we may get unexpected results when some SUTs on this file rely on this method implicitly.

Reply:
I don't think is necessary? Each file is isolated, prefer clear mock only when needed.

Reply:
yes - but unlike other inline-mocks, `getCurrentDatabaseAlterationTimestamp` is globally mocked, and i think that every test case should not assume the original behavior persists if the function has been globally mocked.

in other words, test case should be responsible about the global mocks if needed